frontend/src/pages/modelServing/screens/projects/__test__/utils.spec.ts

```diff
@@ -0,0 +1,115 @@
+import { filterOutConnectionsWithoutBucket } from '~/pages/modelServing/screens/projects/utils';
+import { DataConnection } from '~/pages/projects/types';
+
+describe('filterOutConnectionsWithoutBucket', () => {
+  it('should return an empty array if input connections array is empty', () => {
+    const inputConnections: DataConnection[] = [];
+    const result = filterOutConnectionsWithoutBucket(inputConnections);
+    expect(result).toEqual([]);
+  });
+
+  it('should filter out connections without an AWS_S3_BUCKET property', () => {
+    const inputConnections: DataConnection[] = [
+      {
+        type: 0,
```

Comment:
Same comment as the one in the `mockSecretK8sResource` if we have a mock resource, and we need to parametrize it, just do that, get the mock, add the parameters you need and then create several instances, work with that, not with several built-in objects here.